src/MongoDB.Driver.Core/Core/Configuration/ServerSettings.cs

```diff
@@ -34,7 +35,7 @@ public class ServerSettings
         /// <summary>
         /// Gets the default heartbeat timeout.
         /// </summary>
-        public static TimeSpan DefaultHeartbeatTimeout => TimeSpan.FromSeconds(10);
+        public static TimeSpan DefaultHeartbeatTimeout => Timeout.InfiniteTimeSpan;
```

Comment:
Ideally the default would be null to represent "not provided", but since the field is not nullable we will use Infinity to represent "not provided".

Reply:
Acknowledged